test/client/user_apikey.test.js

```diff
@@ -0,0 +1,258 @@
+      })
+      .catch(e => {
+        fail('Should not reach here');
+      })
+      .then(() => {
+        return th.stitchClient.getApiKeys()
+      })
+      .then(response => {
+        expect(response.length).toEqual(1);
+      })
+      .catch(e => {
+        fail('Should not reach here');
+      });
+  });
+
+  it('can get the user api key by id with element inserted', async() => {
+    let apiID;
+    return th.stitchClient.createApiKey({'name': 'userKey1'})
+      .then(response => {
+        assertApiKey(response, 'userKey1', response._id, false)
+        apiID = response._id;
+      })
+      .then(() => {
+        return th.stitchClient.getApiKeyByID(apiID)
+      })
+      .then(response => {
+        assertApiKey(response, 'userKey1', response._id, false)
+      }).catch(e => {
+        fail('Should not reach here');
+      });
+  });
+
+  it('can disable and endable an user api key', async() => {
+    let apiID;
+    await th.stitchClient.login()
+      .then(async () => {
+        const p1 = await th.stitchClient.createApiKey({'name': 'userKey1'})
+        return p1
+      })
+      .then(async (response) => {
+        assertApiKey(response, 'userKey1', response._id, false)
+        apiID = response._id;
+      })
+      .then(async () => {
+        const p2 = await th.stitchClient.disableApiKeyByID(apiID)
+        return p2
+      })
+      .then(async(res) => {
+        expect(res.status).toEqual(204);
+        const p3 = await th.stitchClient.getApiKeyByID(apiID)
+        return p3
+      })
+      .then(async(res2) => {
+        assertApiKey(res2, 'userKey1', res2._id, true)
+        const p4 = await th.stitchClient.enableApiKeyByID(apiID)
+        return p4
+      })
+      .then(async(res3) => {
+        expect(res3.status).toEqual(204);
+        const p5 = await th.stitchClient.getApiKeyByID(apiID)  
+        return p5
+      })
+      .then(async(res4) => {
+        assertApiKey(res4, 'userKey1', res4._id, false)
+      })
+      .catch(e => {
+        fail('Should not reach here');
+      });
+  });
+
+  it('tests the creating 20 api keys', async() => {
+    let ps = [];
+    for (let i = 0; i < 20; i++) {
+      ps.push(th.stitchClient.createApiKey({'name': `userKey${i}`}));
+      await new Promise(resolve => setTimeout(resolve, 200));
+    }
+    Promise.all(ps)
+      .then(res => {
+        expect(res.length).toEqual(20);
+      })
+      .catch(e => {
+        console.log(e);
+        fail('asdfasdfasdfas');
+      });
+  });
+
+  it('tests creating more than 20 api keys', async() => {
+    let ps = [];
+    for (let i = 0; i < 21; i++) {
+      ps.push(th.stitchClient.createApiKey({'name': `userKey${i}`}));
+      await new Promise(resolve => setTimeout(resolve, 200));
+    }
+    Promise.all(ps)
+      .then(res => {
+        fail('Should not reach here');
+      })
+      .catch(e => {
+        expect(e).toBeInstanceOf(Error);
+        expect(e.response.status).toBe(400);
```

Comment:
would it be useful to write a function `assert400Error` similar to `assert404Error` below?

Reply:
Rewrote `assert404Error` to allow more error types.